idaes/core/unit_model.py

```diff
@@ -156,6 +156,7 @@ def add_port(blk, name=None, block=None, doc=None):
                 slicer = block[:].component(member_list[s].local_name)[...]
 
             r = Reference(slicer)
+            setattr(blk, "_"+s+"_"+name+"_ref", r)
```

Comment:
Any reason `setattr` is used here instead of `blk.add_component`?

Reply:
Only that it took a while for someone to teach me  about `add_component`, and I still forget it exists occasionally.